ldes-server-compaction/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/compaction/application/services/CompactionCandidateService.java

```diff
@@ -34,40 +30,20 @@ public CompactionCandidateService(AllocationRepository allocationRepository, Fra
 	 * @return a structured group of Fragments for a view that can be compacted
 	 */
 	public Collection<Set<CompactionCandidate>> getCompactionTaskList(ViewCapacity viewCapacity) {
-		AtomicInteger index = new AtomicInteger();
-		Map<Integer, Set<CompactionCandidate>> compactionDesign = new HashMap<>();
 		List<CompactionCandidate> possibleCompactionCandidates = getPossibleCompactionCandidates(viewCapacity);
 
 		if (possibleCompactionCandidates.isEmpty()) {
 			return List.of();
 		}
 
-		StreamEx.of(possibleCompactionCandidates.stream())
-				.forPairs((cc1, cc2) -> {
-					if (cc1.getFragment().isConnectedTo(cc2.getFragment())) {
-						Set<CompactionCandidate> set = compactionDesign.getOrDefault(index.get(), new HashSet<>());
-						var totalSum = set.stream().map(CompactionCandidate::getSize).reduce(Integer::sum).orElse(cc1.getSize());
-
-						if (totalSum + cc2.getSize() <= viewCapacity.getCapacityPerPage()) {
-							set.addAll(Set.of(cc1, cc2));
-							compactionDesign.put(index.get(), set);
-						} else {
-							index.incrementAndGet();
-						}
-					}
-				});
-
-		return compactionDesign.values();
```

Comment:
not necessarily saying this is wrong, but it surprises me that all this code can be thrown away?